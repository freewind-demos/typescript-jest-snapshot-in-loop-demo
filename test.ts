import {hello} from "./hello";

test('1+2 === 3', () => {
  expect(1 + 2).toMatchSnapshot();
})

test('a+b === ab', () => {
  expect('a' + 'b').toMatchSnapshot();
})

test('array', () => {
  const array1 = [1, 2];
  const array2 = [3, 4];
  expect([...array1, ...array2]).toMatchSnapshot();
})

test('function', () => {
  expect(hello).toMatchSnapshot();
})

test('snapshot in loop', () => {
  const array = ['aaa', 'bbb', 'ccc'];
  for (const item of array) {
    expect(item).toMatchSnapshot(item);
  }
})
