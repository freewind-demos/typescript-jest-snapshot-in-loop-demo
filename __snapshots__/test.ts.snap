// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1+2 === 3 1`] = `3`;

exports[`a+b === ab 1`] = `"ab"`;

exports[`array 1`] = `
Array [
  1,
  2,
  3,
  4,
]
`;

exports[`function 1`] = `[Function]`;

exports[`snapshot in loop: aaa 1`] = `"aaa"`;

exports[`snapshot in loop: bbb 1`] = `"bbb"`;

exports[`snapshot in loop: ccc 1`] = `"ccc"`;
